fdbserver/worker.actor.cpp

```diff
@@ -3363,7 +3363,14 @@ ACTOR Future<Void> monitorLeaderWithDelayedCandidacy(
 	state Future<Void> monitor = monitorLeaderWithDelayedCandidacyImpl(connRecord, currentCC);
 	state Future<Void> timeout;
 
-	wait(recoveredDiskFiles);
+	// When encryption is enabled, and if the worker is a storage node, it needs to get the cluster controller
+	// interface in order to further obtain the updated EncryptKeyProxy interface to open local storage engine.
+	// Waiting for recoveredDiskFiles here will prevent the worker receive cluster controller interface,
+	// and create a deadlock.
+	// TODO(yiwu): can we remove the wait even when encryption is not enabled?
+	if (!SERVER_KNOBS->ENABLE_ENCRYPTION) {
```

Comment:
@sfc-gh-etschannen not sure if you will have a chance to take a look if it is safe to remove this `recoveredDiskFiles` wait. Though right now I make it an encryption-only change.

Reply:
If I have a tlog with a 500GB disk queue files, recovering that file can take 10+ minutes in extreme cases. If we do not wait for recoveredDiskFiles to be ready, this tlog which is stuck recovering for this long could be recruited as part of the next generation of tlogs.

This was added because we actually had an outage caused by this scenario. It is possible to solve this problem in a more complicated way, by having multiple stages of recovery, which we can discuss offline if this is the only way to solve the deadlock.

Reply:
@sfc-gh-etschannen I think the issue for slow TLog is addressed in #7510, where the worker will only register itself with CC as non-TL or SS role before `recoveredDiskFiles` is finished. The wait here is a fallout from that PR though.